emails/views.py

```diff
@@ -307,23 +307,22 @@ def _index_DELETE(request_data, user_profile):
 @csrf_exempt
 def sns_inbound(request):
     incr_if_enabled("sns_inbound", 1)
-    # We can check for some invalid values in headers before processing body
-    # Grabs message information for validation
-    topic_arn = request.headers.get("X-Amz-Sns-Topic-Arn", None)
-    message_type = request.headers.get("X-Amz-Sns-Message-Type", None)
+    # First thing we do is verify the signature
+    json_body = json.loads(request.body)
+    verified_json_body = verify_from_sns(json_body)
 
-    # Validates header
-    error_details = validate_sns_header(topic_arn, message_type)
+    # Validate ARN and message type
```

Comment:
Future work: It seems to me that this code could be in `_sns_inbound_logic` (which would just take `verified_json_body` or in `verify_from_sns` (which currently just checks the crypto signing). But, there are other changes to make to fully migrate from a push endpoint to the queue pulling task, we can do it then.